File: python/examples/run_searching.py
from dsa_utils import binary_search

def main():
    sorted_data = [2, 5, 7, 13, 21, 42, 64, 89]
    target = 42

    idx = binary_search(sorted_data, target)
    print(f"Dataset: {sorted_data}")
    print(f"Target {target} found at index: {idx}")

if __name__ == "__main__":
    main()

File: rust/dsa-core/examples/run_searching.rs
use dsa_core::searching::binary_search;

fn main() {
    let sorted_data = vec!["apple", "banana", "cherry", "dragonfruit", "elderberry"];
    let target = "dragonfruit";

    match binary_search(&sorted_data, &target) {
        Some(index) => println!("Found '{}' at array index [{}]", target, index),
        None => println!("Element '{}' not found in dataset.", target),
    }
}
